### DISCUSSION TITLE
Clarify the meaning of each "event" for tooltips, and add an "active" event?

### DISCUSSION THREAD

**Discussion by Nantris** [mantinedev/mantine#2229] [Feature requests]
Only by looking at the source code was I able to determine that for `touch` to work, `hover: true` must also be set. And for `focus` to work, the element must be focused via the keyboard, not via a mouse click. 

After learning these particularities, I'm finding that our use case is not covered. What do you think about an event like HTML's `active`, that fires without the element receiving focus, so that tooltips can fire in response to non-focusable button elements.

A contrived example, but quite similar to our actual needs: a rich text editor's toolbar, where it's undesirable to have the button steal focus from the editor, but a user might want to pop up a message when the button is clicked)

Currently, even if the button receives focus from the user's click, `events: { focus: true }` will, counterintuitively, do nothing.

**Comment by rtivital** [comment]
Mantine tooltip is base on floating ui – https://floating-ui.com/docs/react-dom-interactions
Events are based on useHever and useFocus hooks, there is not much control over the behavior – I doubt that it will be easy to add `active` event

**Reply by Nantris** [reply]
Makes sense. I noticed they do have a `useClick` hook. I wonder if that could be used to achieve a similar behavior to allow showing tooltips in response to clicks?

https://floating-ui.com/docs/useClick

**Reply by Nantris** [reply]
Am I wrong to think it could get the job done? I'm not too well versed with Floating UI' yet since we're still on an older version of PopperJS here, but it seems promising to me.

**Reply by rtivital** [reply]
I guess, it should work, don't really know why we need to support it though as I've never seen such a UX pattern.

**Reply by Nantris** [reply]
While I don't have a third-party example handy, our use case is to trigger a tooltip-toast when a button is pressed - for example a code copy button.

I know Mantine has a copy button component but it would be difficult to finagle into our project for a number of reasons, for example #2228. Our copy button also needs to work inside of a `webview` on React Native, where it needs to rely on [non-browser based methods](https://docs.expo.dev/versions/latest/sdk/clipboard/) of loading data into the clipboard.

The tooltip-based method works nicely on desktop and mobile both. Desktop example below:

### Example:

**Hovered:**
![image](https://user-images.githubusercontent.com/6835891/187044985-bb243019-1c57-4792-8e50-11ad82a340e6.png)


**Clicked:**
![image](https://user-images.githubusercontent.com/6835891/187044998-cf5e9d26-bd4a-43d5-ade6-ac15d5a5a62e.png)

**Reply by rtivital** [reply]
You should use controlled tooltip for that, I doubt that useClick implementation of floating UI will meet your requirements anyway.

**Reply by Nantris** [reply]
> You should use controlled tooltip for that

That's fair, although adds a bit of boilerplate (and might add a lot of boilerplate in our case to try to make it operate with custom components.)

PS: Found an example in the wild of the same UX pattern: https://getbootstrap.com/docs/4.0/components/tooltips/#example-enable-tooltips-everywhere

**Reply by Nantris** [reply]
This is proving substantially more difficult to implement via a controlled component than I'd expected.

**Problem 1:**

**Resolution:** I was able to resolve this with the `disabled` prop when the clickable tooltip is showing.

<details>
 <summary>Problem 1</summary>

One very substantial downside of attempting to implement this with a controlled tooltip in conjunction with a hover tooltip is that the hover tooltip will continue to show when a clickable tooltip is triggered.

I tried to rectify this by setting `events: { hover: false }` when the controlled tooltip is opened, but it doesn't work. I'm not sure if that's a limitation related to #2193 or if it's a separate issue.

Here's a repro for the interference of hover/click: https://codesandbox.io/s/thirsty-nobel-ino7mg?file=/src/App.tsx - If you click, you'll see nothing until you stop hovering. Can you provide any guidance?
</details>

----
**Problem 2**

**Resolution** I was able to resolve this by wrapping the inner `Tooltip` in a `span` - which fortunately has not caused any styling or other issues for us. **Although the offsets don't seem quite right somehow, eg `offset: 2` is further from target than `offset: 4`**

<details>
 <summary>Problem 2</summary>

I'm also seeing an issue with positioning of the clickable tooltip not working as intended - but so far I can't reproduce that in a codesandbox:

![image](https://user-images.githubusercontent.com/6835891/187095387-7ddeaeaa-861b-4b66-b1c9-34d82aac27ef.png)

You can see that the outer-Tooltip instance is missing the `position/placement` property, even though we're setting it:

![image](https://user-images.githubusercontent.com/6835891/187095846-876d4454-9e75-47ae-baa0-bce9a62db58f.png)
</details>
